src/compiler/checker.ts

```diff
@@ -22065,6 +22067,19 @@ namespace ts {
                     return Debug.fail();
             }
 
+            function checkTypeComparableWithLessOrGreaterThanOperator(valueType: Type): boolean {
+                const t = valueType.flags;
+                return (t === TypeFlags.String) ||
+                    (t === TypeFlags.Number) ||
+                    (t === TypeFlags.Any) ||
+                    (t === TypeFlags.TypeParameter) ||
+                    (t === TypeFlags.Enum) ||
+                    (t === TypeFlags.Object) ||

```

Comment:
This differs from the PR description and the proposed change in the linked issue. It allows basically any structured type. That means all objects are allowed. That's awesome if you need to compare `Date` objects but may not be desired in all other cases.

Reply:
I have added some more types:
- Object, 
- Intersection, 
- TypeFlags.EnumLiteral | TypeFlags.Union | TypeFlags.UnionOfPrimitiveType
- TypeFlags.Union | TypeFlags.UnionOfPrimitiveTypes
for, at least, TypeScript to continue compiling.

Without the Object type selection, the compilation of TypeScript will fail because Dates are compared somewhere in the code.